examples/aggregation-builder.php

```diff
@@ -48,15 +50,15 @@ function toJSON(object $document): string
         totalCount: Aggregation::sum(1),
         evenCount: Aggregation::sum(
             Aggregation::mod(
-                Expression::fieldPath('randomValue'),
+                Expression::numberFieldPath('randomValue'),
```

Comment:
That's a request from @alcaeus in the Technical Design doc.

> To complete the type experience, there are also typed fieldPath helpers (e.g. stringFieldPath) which communicate to the type system that the field path resolves to the given type. This can again be used by an ORM to verify that the field will actually resolve to the given type (if known).